src/lib/homestore/shard_manager.cpp

```diff
@@ -1,15 +1,199 @@
 #include "homeobject.hpp"
+#include "replication_message.hpp"
+
+#include <homestore/homestore.hpp>
+#include <homestore/meta_service.hpp>
 
 namespace homeobject {
 
 uint64_t ShardManager::max_shard_size() { return Gi; }
+uint64_t ShardManager::max_shard_num_in_pg() {return ((uint64_t)0x01) << 48;}
+
+bool HSHomeObject::check_if_pg_exist(pg_id pg) const {
```

Comment:
Can we follow the pattern for `_get_shard` and do a `_get_pg` that returns the `Result<PG>` itself? The user doesn't have to consume it, but makes the code consistent for anything that looks up pgs.

Reply:
good point, it is added now.